### ISSUE TITLE
Crash if bbswitch module is not installed

### ISSUE THREAD

**Issue by Krulvis** [Askannz/optimus-manager#133]
When using the command `optimus-manager --switch nvidia` it gets stuck in tty1.  Switching to other tty and back doesn't resolve the issue.

**System info**
Please include :
distro: Archlinux
desktop manager: Gnome
Display manager: lightDM
Laptop model: Clevo N850HK
optimus-manager: 1.1

**Logs**
[optimus-manager-journal.txt](https://github.com/Askannz/optimus-manager/files/3635275/optimus-manager-journal.txt)

**Comment by Krulvis** [comment]
I am using the options:

[optimus]
auto_logout=yes
pci_power_control=no
switching=none

**Comment by Mario156090** [comment]
> I am using the options:
> 
> [optimus]
> auto_logout=yes
> pci_power_control=no
> switching=none

Hi, you should set a switching option. For example, you can use:
bbswitch
acpi_call

If you want to use bbswitch first install bbswitch package.

pacman -S bbswitch

for more information you can read /usr/share/optimus-manager.conf

**Comment by lan542662** [comment]
> > I am using the options:
> > [optimus]
> > auto_logout=yes
> > pci_power_control=no
> > switching=none
> 
> Hi, you should set a switching option. For example, you can use:
> bbswitch
> acpi_call
> 
> If you want to use bbswitch first install bbswitch package.
> 
> pacman -S bbswitch
> 
> for more information you can read /usr/share/optimus-manager.conf

thank you. May i know if i need configure bbswitch, i use archlinux, thank you in advanced.

**Comment by Mario156090** [comment]
Default config works very well with bbswitch.

**Comment by Krulvis** [comment]
changing to bbswitch didn't do it for me. I couldn't even get the display manager to work on boot. However changing configs to boot with nvidia did the trick for me. I can switch fine between nvidia and intel now using `switching=none` as long as I boot with nvidia.

**Comment by Mario156090** [comment]
> =

Ok, then you fix it.

**Comment by Krulvis** [comment]
If someone is interested I can still attempt to get the journalctl output for the problem that I have when using `switching=bbswitch` so it can be looked at for other people, otherwise I am just gonna be happy that it worked out for me this way.

**Comment by yaliv** [comment]
It seems that installing bbswitch is "required", although I don't change the default setting (`switching=none`) and I think my laptop and the Nvidia driver (435.21) are capable of handling this.

I got this error when trying to switch to nvidia or hybrid:
```sh
Unloading Nvidia modules (if any)
Unloading nouveau module (if any)
Unloading bbswitch module (if any)
ERROR : Nvidia PCI reset failed. Continuing. Error is : Cannot unload bbswitch : Failed to execute 'modprobe -r bbswitch' : modprobe: FATAL: Module bbswitch not found.
Unloading bbswitch module (if any)
Cannot setup GPU : kernel setup error : Cannot unload bbswitch : Failed to execute 'modprobe -r bbswitch' : modprobe: FATAL: Module bbswitch not found.
```

It doesn't even create any file in `/etc/X11/xorg.conf.d`.

I have kernel 5.3 and 5.2 installed and running on 5.3.
So when I choose to install `linux53-bbswitch` package, `linux52-bbswitch` will be selected as well.

Now the switching works:
```sh
Unloading Nvidia modules (if any)
Unloading nouveau module (if any)
Unloading bbswitch module (if any)
Unloading bbswitch module (if any)
Running /etc/optimus-manager/nvidia-enable.sh
Running /etc/optimus-manager/nvidia-disable.sh
Loaded extra Intel Xorg options (0 lines)
Loaded extra AMD Xorg options (0 lines)
Loaded extra Nvidia Xorg options (0 lines)
Writing to /etc/X11/xorg.conf.d/10-optimus-manager.conf
```

**Comment by Siege0x** [comment]
It works well when I used optimus-manager: 1.1, switching=none and I had installed the gdm-prime and so on,but it doesn't work in 1.2?is it the 'none' means the using 'prime' as switch methods?

**Comment by lan542662** [comment]
> Default config works very well with bbswitch.

Thank you, i think so. I have another question. Archlinux prefers use modesetting rather than xf86-video-intel , what to you think about this. Thanks.. I just use archlinux a short time.
...
[intel]
#modesetting, intel
driver=modesetting

**Comment by wizonesolutions** [comment]
I had missed installing `linux53-bbswitch` as well. That was the cause of this issue for me.

**Comment by Askannz** [comment]
Yep, the program crashes if bbswitch isn't installed. That's totally a bug, fix is on the way.

**Comment by Askannz** [comment]
Fixed in 68928761d98ea12d92320035e47333909c3aa517
